spec/controllers/users_controller_spec.rb

```diff
@@ -130,6 +130,55 @@
         expect(flash.now[:errors]).not_to be_nil
       end
     end
+
+    describe '#show_update_email' do
+      it 'renders the update email page' do
+        get :show_update_email, :params => { :user_id => user_id }
+
+        expect(response).to have_http_status(:success)
+        expect(subject).to render_template(:show_update_email)
+      end
+    end
+
+    describe '#update_email' do
+      it 'updates the user email address' do
+        stub_cognito_response(method: :admin_update_user_attributes, payload: {})
+        post :update_email, :params => { update_user_email_form: { email: "test1@test.com"}, user_id: user_id}
+        expect(subject).to redirect_to(update_user_path)
+      end
+
+      it 'fails with error when form not valid' do
+        stub_cognito_response(method: :admin_update_user_attributes, payload: {})
+        post :update_email, :params => { update_user_email_form: { email: ""}, user_id: user_id}
+        expect(response).to have_http_status(:bad_request)
+        expect(subject).to render_template(:show_update_email)
+      end
+
+      it 'fails with error when form not valid' do
+        form = UpdateUserEmailForm.new({ email: ""})
+        form.valid?
```

Comment:
is this affecting the test in any way? it's defined outside of the controller so I don't think it is

Reply:
No its only there to replicate what should happen in the controller so I can get the errors out of it.  I can rename the variable to make that clearer if that would help?

Reply:
ah ok, would probably be better to use the actual value in the expectation instead of recreating the model in the test, but ok...

Reply:
I have changed test to test specific errors instead of creating a new object just for the error message assertions.